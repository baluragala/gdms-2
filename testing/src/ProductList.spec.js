import React from "react";
import ProductList from "./ProductList";
import renderer from "react-test-renderer";

describe("Product List", () => {
  it("should have initial state with 4 products", () => {
    const tree = renderer.create(<ProductList />);
    const testInstance = tree.root;
    const initialState = {
      products: ["iphone", "pixel ", "note", "oneplus 6"]
    };
    expect(testInstance.instance.state).toMatchObject(initialState);
  });

  it("should render 3 products", () => {
    const tree = renderer.create(<ProductList />);
    const testInstance = tree.root;
    expect(testInstance.findAllByType("li").length).toEqual(4);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should have called alert on button click", () => {
    // const tree = renderer.create(<ProductList />);
    // const testInstance = tree.root;
    // const button = testInstance.findByType("button");
    // console.log(button);
    const mock = jest.fn();
    mock.mockReturnValue("HELLO");
    let status = mock("hello");
    expect(status).toBeUndefined();
    expect(mock).toHaveBeenCalled();
    expect(mock).toHaveBeenCalledTimes(1);
    expect(mock).toHaveBeenCalledWith("Hello");
  });

  it("async with done", done => {
    setTimeout(done, 100);
  });

  it("async test with prmoise", () => {
    return new Promise((resolve, reject) => setTimeout(resolve, 100));
  });

  it("async test with async/await", async () => {
    console.log(1);
    await new Promise((resolve, reject) => {
      setTimeout(() => {
        console.log(2);
        resolve();
      }, 100);
    });
    console.log(3);
  });
});
