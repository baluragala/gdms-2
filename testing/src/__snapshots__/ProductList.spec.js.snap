// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List should render 3 products 1`] = `
<div>
  <ul>
    <li>
      iphone
    </li>
    <li>
      pixel 
    </li>
    <li>
      note
    </li>
    <li>
      oneplus 6
    </li>
  </ul>
  <button
    onClick={[Function]}
  >
    Click here
  </button>
</div>
`;
